charts/seed-bootstrap/templates/gardener-seed-admission-controller/validatingwebhookconfiguration.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: gardener-seed-admission-controller
+  labels:
+    app: gardener
+    role: seed-admission-controller
+webhooks:
+  - name: seed.admission.core.gardener.cloud
+    rules:
+    - apiGroups:
+      - apiextensions.k8s.io
+      apiVersions:
+      - v1alpha1
+      - v1beta1
+      - v1
+      operations:
+      - DELETE
+      resources:
+      - customresourcedefinitions
+    - apiGroups:
+      - extensions.gardener.cloud
+      apiVersions:
+      - v1alpha1
+      operations:
+      - DELETE
+      resources:
+      - backupbuckets
+      - backupentries
+      - controlplanes
+      - extensions
+      - infrastructures
+      - networks
+      - operatingsystemconfigs
+      - workers
```

Comment:
> I assume, you left out the Cluster resource on purpose, because no one should act upon it, right?

Right.

> So I guess, this is also something worth protecting.

The purpose of the protection is prevent catastrophic situations. Deleting the `Cluster` resource will only make the reconciliation fail as it only serves as  kind of configmap. Hence, no need to protecting it.

> Also what about the CRs in dns.gardener.cloud? You marked the CRDs as deletion protected but do not check the CRs here.

On purpose as well. If we would protect it we would have to change the DNS controller manager as it deletes the DNS entries. We don't want this at this point in time. Hence, I only protected the CRD for now. I'm open to later also protect the custom resources as well in case the DNS controller manager adopts the deletion confirmation logic. But I don't want to depend on it now.

> Though, this would not be so hard-to-recover-from as with the other resources, because it can self-heal on the next reconciliation.

Right.

> I think we should also protect the ManagedResource CRD/CRs, because we use it to deploy all the extension controllers and also some extensions are depending on it to deploy stuff into seed/shoot which might be critical.

Agreed to protect the CRD itself, will do. However, I won't protect the CRs as it again would require adoption in all the extension controllers (= not backwards compatible). Same rationale like for the DNS objects.

Reply:
> The purpose of the protection is prevent catastrophic situations. Deleting the `Cluster` resource will only make the reconciliation fail as it only serves as kind of configmap. Hence, no need for protecting it.

Ok, fair enough.

> If we would protect it we would have to change the DNS controller manager as it deletes the DNS entries. We don't want this at this point in time.

Oh yeah, got it. I didn't consider the implication on the dns controller manager's own deletion logic.

> However, I won't protect the CRs as it again would require adoption in all the extension controllers (= not backwards compatible).

Same here, also didn't think about the deletion logic in the controllers. But agreed, protecting the dns/managedresource CRDs is already a good measure to prevent catastrophic situations 👍